packages/utils/src/mergeDeep.ts

```diff
@@ -12,13 +12,38 @@ export function mergeDeep<S extends any[]>(
   respectArrays = false,
   respectArrayLength = false,
 ): UnboxIntersection<UnionToIntersection<BoxedTupleTypes<S>>> & any {
-  const target = sources[0] || {};
+  if (respectArrays && respectArrayLength) {
+    let expectedLength: number | undefined;
+    const areArraysInTheSameLength = sources.every(source => {
+      if (Array.isArray(source)) {
+        if (expectedLength === undefined) {
+          expectedLength = source.length;
+          return true;
+        } else if (expectedLength === source.length) {
+          return true;
+        }
+      }
+      return false;
+    });
+
+    if (areArraysInTheSameLength) {
+      return new Array(expectedLength).fill(null).map((_, index) =>
```

Comment:
Why creating a new Array ? We could just map over the first source is they have all the same length ? :-)

Reply:
TypeScript fun :D When I do `sources[0]`, it doesn't see it as an array so I have to cast it